pvlib/pvsystem.py

```diff
@@ -1887,6 +1887,87 @@ def sapm_celltemp(poa_global, wind_speed, temp_air,
     return pd.DataFrame({'temp_cell': temp_cell, 'temp_module': temp_module})
 
 
+def pvsyst_celltemp(poa_global, wind_speed, temp_air, eta_m=0.1,
+                    alpha_absorption=0.9, temp_model="freestanding"):
+    """
+    Calculate cell temperature using the PVSyst model.
+
+    Parameters
+    ----------
+    poa_global : numeric
+        Total incident irradiance in W/m^2.
+
+    wind_speed : numeric
+        Wind speed in m/s at a height of 10 meters.
+
+    temp_air : numeric
+        Ambient dry bulb temperature in degrees C.
+
+    eta_m : numeric
+        Module external efficiency as a fraction, i.e., DC power / poa_global.
+
+    alpha_absorption : float
+        Absorption coefficient, default is 0.9.
+
+    temp_model : string, tuple, or list, default 'freestanding' (no dict)
+        Model to be used.
+
+        If string, can be:
+
+            * 'freestanding' (default)
+                Modules with rear surfaces exposed to open air (e.g. rack
+                mounted).
+            * 'insulated'
+                Modules with rear surfaces in close proximity to another
+                surface (e.g. roof mounted).
+
+        If tuple/list, supply parameters in the following order:
+
+            * natural_convenction_coeff : float
+                Natural convection coefficient. Freestanding default is 29,
+                fully insulated arrays is 15.
+
+            * forced_convection_coeff : float
+                Forced convection coefficient, default is 0.
+
+    Returns
+    -------
+    temp_cell : numeric or Series
+        Cell temperature in degrees Celsius
+
+    References
+    ----------
+    [1]"PVsyst 6 Help", Files.pvsyst.com, 2018. [Online]. Available:
+    http://files.pvsyst.com/help/index.html. [Accessed: 10- Dec- 2018].
+
+    [2] Faiman, D. (2008). "Assessing the outdoor operating temperature of
+    photovoltaic modules." Progress in Photovoltaics 16(4): 307-315.
+    """
+
+    temp_models = {"freestanding": (29.0, 0), "insulated": (15.0, 0)}
+
+    if isinstance(temp_model, str):
+        natural_convenction_coeff, forced_convection_coeff = temp_models[
+            temp_model.lower()
+        ]
+    elif isinstance(temp_model, (tuple, list)):
```

Comment:
@camerontstark in the future maybe consider [duck typing](http://www.voidspace.org.uk/python/articles/duck_typing.shtml#problem) instead of type checking.  Even though duck typing is the Pythonic preference, it may not always work best. I think your solution here is fine, but in general I think duck typing usually results in more robust code. Eg:
```python
else:
    natural_convenction_coeff, forced_convection_coeff = temp_model
    # already raises ValueError or TypeError
    # or use try: except to raise a more descriptive exception
```

Reply:
@mikofski Yes, thanks for the suggestion! I've been trying to pay attention more to duck typing in my code and thought about it for this instance but since this was my first contribution and  sapm_celltemp() had the Look Before You Leap type check in it already I figured I'd just follow convention.